### PR TITLE
Do not print trailing commas in arrow function type parameter lists in `ts` code blocks

### LINKED ISSUE #15282: Prettier prints a trailing comma after type argument list even if the languages is `ts` in markdown
**Prettier 3.0.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEADdMDOAdKkqYwAExAvEQDwAqANAHwAUAlEaXUcAL4Dcu6quPhkwAPXPkIlWlWoxZsOnIahA0QEAA4wAltEzJQAQwBOxiAHcACiYT6UhgDbnDAT31qARscNgA1nBgAZUMAWzgAGW0oOGQAM0dMOE9vPwDAjR8ogHNkGGMAVySQRJDtXIKizGyHOABFfIh4OISigCtRQOq6hqakeIdEtQBHHrhLMw07EENMAFpouAATJdUQPMNtB2yAYQgQkMNkaYcHVaqoLJqAQRg87Q98+Es4Y0jo5oGigAsYEIcAdS+2ngmAyYDggVswO0ADdgS4jmBMO4QDDCgBJKDLWCBMDGbRaK5YwIwFw1D6DEAaMyJf7eDRHalwRLGGExNRRFkwcaGLIHClFDLGFlHA7GXyLCxQVbUqIwf7aRYwL7IAAcAAY1MY4CNtNqeXzDn0WmoYIYPAqlSqkAAmNT5RJUc12fqUuAhDxLZaLcKGC75XlwABiEGMB1u2SOhkeEBAnE4QA)

```sh
--parser markdown
```

**Input:**

````markdown
```ts
const t = <T,>() => {};
```

```tsx
const t = <T,>() => {}
```
````

**Output:**

````markdown
```ts
const t = <T,>() => {};
```

```tsx
const t = <T,>() => {}
```

````

The second codeblock prints a trailing comma and it's the correct behavior, however the first codeblock should remove the trailing comma

**Expected behavior:**
The first codeblock should not have a trailing comma

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

Fixes #15282 

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-markdown/embed.js [hunk 1]
```diff
@@ -17,8 +17,13 @@ function embed(path, options) {
...
         const newOptions = { parser };
-        if (node.lang === "tsx") {
-          newOptions.filepath = "dummy.tsx";
+
+        // Override the filepath option.
+        // This is because whether the trailing comma of type parameters
+        // should be printed depends on whether it is `*.ts` or `*.tsx`.
+        // https://github.com/prettier/prettier/issues/15282
+        if (node.lang === "ts" || node.lang === "tsx") {
+          newOptions.filepath = `dummy.${node.lang}`;
```
Comment (fisker) [inline src/language-markdown/embed.js:26]: How should we print this?

````md
```typescript
const foo = <T>() => {}
```
````

**Comment by fisker** [comment]
https://github.com/prettier/prettier/blob/999c2c27506735f67dfd22e0d5d6c381dda209ee/src/language-js/parse/typescript.js#L43


This link is broken, can you fix it? https://typescript-eslint.io/packages/parser/#jsx

File: src/language-markdown/embed.js (hunk 1 above)
Comment (sosukesuzuki) [reply src/language-markdown/embed.js:26]: Same as for `ts`

Comment (sosukesuzuki) [reply src/language-markdown/embed.js]: Fixed

**Review by fisker — APPROVED** [review]

**Review by fisker — APPROVED** [review]

File: tests/format/markdown/code/typescript-trailing-comma.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+```typescript
```
Comment (fisker) [inline tests/format/markdown/code/typescript-trailing-comma.md:1]: Better put two  blocks in one file.

Comment (fisker) [reply tests/format/markdown/code/typescript-trailing-comma.md:1]: Maybe "TypeScript" lang should also tested?

Comment (sosukesuzuki) [reply tests/format/markdown/code/typescript-trailing-comma.md:1]: No, we don't format `TypeScript`:

**Prettier 3.0.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEADdAVAngBzgZTACcBLHGAHSgEYACegalrvtoG4r1UrP0Zc4AZ2JlKNVkxb0OULiAA0ICORLRByUAEMiRCAHcACtoTqUmgDZ7NWdYoBGRTWADWcGPk0BbOABkSUOGQAMwtBOHtHFzd8HCd-AHNkGCIAV3CQMM8SJNT0wQTzOABFFIh4YND0gCtBAA98AuLS8qQQ8zDFAEdmuANdHFMQTUEAWgC4ABNJhRBkzRJzBIBhCE9PTWQh83MZ-Kh4woBBGGSSOxT4AzgiPwCK9vSACxhPcwB1R5J4QViwAhMviQAG5fLCbMCCWwgIFpACSUCmsEIpHIhwR+H4hXuHRAOF0YTejhwmzxQmuQMCin8YSIMD6mni62x6ViRBpm3WRGcE30UBmeP8MDeJAmMEeyAAHAAGRREODdEhy+mMjatSqKGCaOzC0XipAAJkUKTCGC1pjaOLgnjskymEx8mn2KQZcAAYhAiOsTglNpoLhAQABfQNAA)

```sh
--parser markdown
```

**Input:**

````markdown
```TypeScript
1   + 1    ;
```

```typescript
1   + 1    ;
```
````

**Output:**

````markdown
```TypeScript
1   + 1    ;
```

```typescript
1 + 1;
```

````

**Comment by fisker** [comment]
If you agree, we should update this line instead https://github.com/prettier/prettier/blob/90fab608aec75214b3ce816e9c317b32cc9314f6/src/language-js/print/type-parameters.js#L41

**Pull request merged**